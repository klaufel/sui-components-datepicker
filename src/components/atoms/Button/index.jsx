import React from 'react'
import PropTypes from 'prop-types'
import './index.scss'

const Button = ({onClick, children, ...props}) => (
  <button className="Button Button--default" onClick={onClick} {...props}>
    {children}
  </button>
)

Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired
}

export default Button
