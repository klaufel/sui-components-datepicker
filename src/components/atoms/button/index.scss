@import '../../../styles/lib/index';

$Button-color: $c-white !default;
$Button-backgroundColor: $c-primary !default;
$Button-backgroundColor--hover: darken($Button-backgroundColor, 10%) !default;
$Button-backgroundColor--active: darken($Button-backgroundColor, 5%) !default;
$Button-borderRadius: $bdrs-none !default;
$Button-PaddingY: $p-v !default;
$Button-PaddingX: $p-h !default;

.Button {
  align-items: center;
  border: 0;
  border-radius: $Button-borderRadius;
  cursor: pointer;
  display: flex;
  font-family: inherit;
  line-height: $lh-button;
  outline: none;
  padding: $Button-PaddingY $Button-PaddingX;
  background-color: $Button-backgroundColor;
  color: $Button-color;

  &:hover {
    background-color: $Button-backgroundColor--hover;
  }

  &:active {
    background-color: $Button-backgroundColor--active;
  }

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }
}
