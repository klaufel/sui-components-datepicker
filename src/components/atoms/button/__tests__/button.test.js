import React from 'react'
import {render} from '@testing-library/react'
import Button from '..'

describe('Button', () => {
  test('renders the component', () => {
    const {container} = render(<Button>Default button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
