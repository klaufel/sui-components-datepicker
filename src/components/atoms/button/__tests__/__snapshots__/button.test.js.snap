// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component 1`] = `
<button
  class="Button Button--default"
>
  Default button
</button>
`;
